dataclasses_avroschema/main.py

```diff
@@ -128,13 +129,40 @@ def _reset_parser(cls: Type["AvroModel"]) -> None:
         cls._parent = None
 
     @classmethod
+    @overload
     def deserialize(
-        cls: Type["AvroModel"],
+        cls: Type[TSelf],
         data: bytes,
-        serialization_type: serialization.SerializationType = "avro",
-        create_instance: bool = True,
-        writer_schema: Optional[Union[JsonDict, Type["AvroModel"]]] = None,
-    ) -> Union[JsonDict, "AvroModel"]:
+        serialization_type: serialization.SerializationType = ...,
+        create_instance: Literal[False] = ...,
```

Comment:
It seems that the order is not correct. We should have `create_instance: Literal[True] = ...,` first to match the default behavior. 

Now I see the following (Default `JsonDict`):

<img width="728" height="480" alt="Screenshot 2025-09-07 at 14 08 02" src="https://github.com/user-attachments/assets/29f5f41e-ef8f-4cc7-a9e0-3429c2b3bec8" />

Reply:
You're missing the `create_instance=True`, when I set it I see `TestEvent` as the return type.

Reply:
Fixed.
For some reason I thought `False` was the default